src/osd/PGLog.h

```diff
@@ -154,7 +154,7 @@ struct PGLog : DoutPrefixProvider {
     // recovery pointers
     std::list<pg_log_entry_t>::iterator complete_to; // not inclusive of referenced item
     version_t last_requested = 0;               // last object requested by primary
-
+    bool can_roll_forward = false; // updates when new entries are added
```

Comment:
No reason to make this a member variable of PGLog.  Can we simply make it a parameter of append_log_entries_update_missing?

Reply:
@athanatos updated, added roll_forward_to version checking using std::optional, what do you think about this approach?

Reply:
I actually meant to add an out param to append_log_entries_update_missing and do the rollforward in merge_log as before.  You've added the rollforward behavior to append_log_update_missing directly, however, which seems fine, but it means you don't need the param at all (all users pass std::nullopt?)

Reply:
yes, that makes sense, updated to straightforward use a bool value instead, as out param for merge_log like keep the legacy flow, @neha-ojha later suggested